### PR TITLE
Sparse Ops Cleanup

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is in preparation for implementing the direct-pinning discussed [here](https://github.com/pandas-dev/pandas/pull/19649#discussion_r167527010).  The main roadblock to that is SparseSeries which calls ops.add_special_arithmetic_methods twice.  This PR gets rid of the double-call, so the follow-up will be straight-forward.

A handful of cleanups are done here too:
 - unrelated docstring fix for TimedeltaIndex.replace
 - minor formatting improvements
 - remove an unused kwarg for DataFrame._compare_frame
 - remove a loop in add_flex_arithmetic_methods, replace with an assertion that it is unnecessary
 - fix names for some SparseArray method names, e.g. ATM `SparseArray.__eq__.__name__ == 'eq'`

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -1080,19 +1078,19 @@ def na_op(x, y):
...
         return result
 
+    fill_int = lambda x: x.fillna(0)
+    fill_bool = lambda x: x.fillna(False).astype(bool)
+
     def wrapper(self, other):
         is_self_int_dtype = is_integer_dtype(self.dtype)
 
-        fill_int = lambda x: x.fillna(0)
-        fill_bool = lambda x: x.fillna(False).astype(bool)
```
Comment (jbrockmendel) [inline pandas/core/ops.py]: Moving these outside of the function scope to clarify they don't need to be re-defined on each call.  _bool_method_SERIES is the topic of a separate PR, but this is sufficiently small to include in assorted-cleanups.

File: pandas/core/ops.py [hunk 2]
```diff
@@ -1568,7 +1562,7 @@ def wrapper(self, other):
                 dtype = getattr(other, 'dtype', None)
                 other = SparseArray(other, fill_value=self.fill_value,
                                     dtype=dtype)
-            return _sparse_array_op(self, other, op, name)
+            return _sparse_array_op(self, other, op, name, series=False)
```
Comment (jreback) [inline pandas/core/ops.py]: this whole sparse interaction is pretty hacky (I know not addressing this now), but....

File: pandas/core/ops.py [hunk 3]
```diff
@@ -1591,4 +1583,5 @@ def wrapper(self, other):
 
 sparse_series_special_funcs = dict(arith_method=_arith_method_SPARSE_SERIES,
                                    comp_method=_arith_method_SPARSE_SERIES,
-                                   bool_method=None)
+                                   bool_method=_bool_method_SERIES)
```
Comment (jreback) [inline pandas/core/ops.py]: can we hold off adding till then

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/ops.py]: This is necessary to get rid of the double-call in sparse.series.

File: pandas/core/ops.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/ops.py]: Yah, there are a couple of improvements in store for the pass after next (which is just implementing the single-call `ops.add_whatever_methods(cls)`)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**